core/src/main/java/io/questdb/cairo/sql/async/PageFrameDispatchJob.java

```diff
@@ -0,0 +1,200 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql.async;
+
+import io.questdb.MessageBus;
+import io.questdb.cairo.sql.NetworkSqlExecutionCircuitBreaker;
+import io.questdb.cairo.sql.PageAddressCacheRecord;
+import io.questdb.cairo.sql.SqlExecutionCircuitBreaker;
+import io.questdb.cairo.sql.SqlExecutionCircuitBreakerConfiguration;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.Job;
+import io.questdb.mp.MCSequence;
+import io.questdb.mp.MPSequence;
+import io.questdb.mp.RingQueue;
+import io.questdb.std.Misc;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.LockSupport;
+
+public class PageFrameDispatchJob implements Job, Closeable {
+
+    private final static Log LOG = LogFactory.getLog(PageFrameDispatchJob.class);
+
+    private final MessageBus messageBus;
+    private final MCSequence dispatchSubSeq;
+    private final RingQueue<PageFrameDispatchTask> dispatchQueue;
+    // work stealing records
+    private final PageAddressCacheRecord[] records;
+    private final SqlExecutionCircuitBreaker[] circuitBreakers;
+
+    public PageFrameDispatchJob(
+            MessageBus messageBus,
+            int workerCount,
+            @Nullable SqlExecutionCircuitBreakerConfiguration sqlExecutionCircuitBreakerConfiguration) {
+        this.messageBus = messageBus;
+        this.dispatchSubSeq = messageBus.getPageFrameDispatchSubSeq();
+        this.dispatchQueue = messageBus.getPageFrameDispatchQueue();
+
+        this.records = new PageAddressCacheRecord[workerCount];
+        this.circuitBreakers = new SqlExecutionCircuitBreaker[workerCount];
+        for (int i = 0; i < workerCount; i++) {
+            records[i] = new PageAddressCacheRecord();
+            if (sqlExecutionCircuitBreakerConfiguration != null) {
+                circuitBreakers[i] = new NetworkSqlExecutionCircuitBreaker(sqlExecutionCircuitBreakerConfiguration);
+            } else {
+                circuitBreakers[i] = NetworkSqlExecutionCircuitBreaker.NOOP_CIRCUIT_BREAKER;
+            }
+        }
+    }
+
+    public static boolean stealWork(
+            RingQueue<PageFrameReduceTask> queue,
+            MCSequence reduceSubSeq,
+            PageAddressCacheRecord record,
+            SqlExecutionCircuitBreaker circuitBreaker
+    ) {
+        if (PageFrameReduceJob.consumeQueue(queue, reduceSubSeq, record, circuitBreaker)) {
+            LockSupport.parkNanos(1);
```

Comment:
Should be `Os.pause()`

Reply:
fixed https://github.com/questdb/questdb/pull/1732/commits/edcb15659325bda39ad82b2f68e46b5dac3d2dd8